src/core/network.js

```diff
@@ -115,20 +133,17 @@ var NetworkManager = (function NetworkManagerClosure() {
         pendingRequest.expectedStatus = 200;
       }
 
-      if (args.onProgressiveData) {
-        // Some legacy browsers might throw an exception.
-        try {
-          xhr.responseType = 'moz-chunked-arraybuffer';
-        } catch(e) {}
-        if (xhr.responseType === 'moz-chunked-arraybuffer') {
-          pendingRequest.onProgressiveData = args.onProgressiveData;
-          pendingRequest.mozChunked = true;
-        } else {
-          xhr.responseType = 'arraybuffer';
-        }
+//#if (GENERIC || FIREFOX || MOZCENTRAL)
+      if (args.onProgressiveData && supportsMozChunked) {
+        xhr.responseType = 'moz-chunked-arraybuffer';
+        pendingRequest.onProgressiveData = args.onProgressiveData;
+        pendingRequest.mozChunked = true;
       } else {
         xhr.responseType = 'arraybuffer';
       }
+//#else
```

Comment:
duplication of code might be fine in this case. okay, may I suggest a compromise that will remove the test for MOZCENTRAL/CHROMIUM, however 'moz-chunked-arraybuffer' branch will still be present?

```
   var useMozChunkedLoading = false;
//#if (FIREFOX || MOZCENTRAL)
// useMozChunkedLoading = !!args.onProgressiveData;
//#endif
//#if !(CHROMIUM || FIREFOX || MOZCENTRAL)
  useMozChunkedLoading = supportsMozChunked && !!args.onProgressiveData;
//#endif
  if (useMozChunkedLoading) {
```